frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -162,6 +208,55 @@ export const TextField = ({
     );
   }
 
+  if (isAutoCompleteable) {
+    const [autoCompleteOptions, setAutoCompleteOptions] = React.useState([]);
+    const autoCompleteDebounce = React.useRef(
+      _.debounce(value => {
+        autocompleteCallback(value)
+          .then(data => {
+            setAutoCompleteOptions(data.results);
+          })
+          .catch(err => {
+            helpText = err;
+          });
+      }, 500)
+    ).current;
+
+    // TODO (mcutalo): support option.label in the renderOption
+    return (
+      <Autocomplete
+        freeSolo
+        size="small"
+        options={autoCompleteOptions}
+        PopperComponent={renderPopper}
+        getOptionLabel={option => (option.id ? option.id : option)}
+        onInputChange={(e, v, r) => autoCompleteDebounce(v)}
+        renderOption={option => (
+          <ResultGrid container alignItems="center">
+            <Grid item xs>
+              <ResultLabel>{option.id}</ResultLabel>
```

Comment:
Is the option guaranteed to have an `id`? the `getOptionLabel` prop accounts for the fact it might be missing. How should we handle that here?

Reply:
Yeah the `id` should always be present, with the `label` being optional.  Which I have yet to add label support quite yet, I plan to do that in a follow up PR as there is a bit more backend work needed.